README.md

```diff
@@ -189,26 +189,29 @@ const { flags, values, positionals } = parseArgs({ args, options });
   - `"0o22"`
 - Does it coerce types?
   - no
-- Does `--no-foo` coerce to `--foo=false`?  For all flags?  Only boolean flags?
-  - no, it sets `{args:{'no-foo': true}}`
+- Does `--no-foo` coerce to `--foo=false`?  For all options?  Only boolean options?
+  - no, it sets `{values:{'no-foo': true}}`
 - Is `--foo` the same as `--foo=true`?  Only for known booleans?  Only at the end?
-  - no, `--foo` is the same as `--foo=`
+  - no, `--foo` is a boolean option and `--foo=true` is a string option
 - Does it read environment variables?  Ie, is `FOO=1 cmd` the same as `cmd --foo=1`?
   - no
 - Do unknown arguments raise an error?  Are they parsed?  Are they treated as positional arguments?
   - no, they are parsed, not treated as positionals
-- Does `--` signal the end of flags/options?
-  - **open question**
-  - If `--` signals the end, is `--` included as a positional?  is `program -- foo` the same as `program foo`?  Are both `{positionals:['foo']}`, or is the first one `{positionals:['--', 'foo']}`?
+- Does `--` signal the end of options?
+  - yes
+- Is `--` included as a positional?
+  - no
+- Is `program -- foo` the same as `program foo`?
+  - yes, both store `{positionals:['foo']}`
```

Comment:
This seems wrong; — means “stop parsing options”; everything after it should be untouched.

Reply:
I'm not sure what you mean. 

The QA don't include an example storing an (otherwise) option as a positional, is that what you were expecting?

Reply:
What i mean is, `program a b c` should have three positionals (a, b, and c) - but `program -- a b c` should only have one positional, the string "a b c", so it can be passed verbatim to something else.

Reply:
No, combining the strings like that would lose the information about the separate arguments and require another interpretation step to split them up again.

This might be too opaque, but imagine implementing subcommands:

```
// utility --option1 -- dispatch --option2
const result1 = parseArgs({ options: mainOptions );
if (result1.positionals[0] === 'dispatch') {
   const result2 = parseArgs({ args: result1.positionals.slice(1), options: dispatchOptions });
}
```

A different example. The `--` turns off option processing and the remaining arguments end up untouched and treated as positionals, so these three examples all have the same result. There isn't anything special about the handling of arguments after the `--` than before the `--` other than no option processing.

```
utility foo bar -- --unprocessed-option
utility foo -- bar --unprocessed-option
utility -- foo bar --unprocessed-option
```

Reply:
Yes, that's the idea. They're going to be passed to another program, so THAT program will be interpreting them on its own - perhaps with different rules than the current one.

Reply:
Normally that other program will be interpreting an array of arguments, not a single string?

The `--` stops option detection. It is not a marker for separate collection of arguments into another structure. 

(There are some programs that treat `--` as a marker for separate processing, but in my experience that is unusual. It is easy to form the impression programs work that way though, since for easy reading the `--` is often put before the passthrough arguments!)

Reply:
One way of thinking about `--` is that it is another user-level intervention. No author action required.
- suppress shell interpretation of special characters in arguments: quote argument
- suppress utility interpretation of arguments as options: `--`

Reply:
Example behaviours of reference implementations added to: https://github.com/pkgjs/parseargs/issues/58#issuecomment-1066017114